packages/native-stack/src/views/HeaderConfig.tsx

```diff
@@ -168,9 +168,10 @@ export default function HeaderConfig({
             ? 'transparent'
             : colors.card)
         }
-        backTitle={headerBackTitleVisible ? headerBackTitle : ' '}
```

Comment:
This change won't be backward compatible, right?

Reply:
On the other hand there's one trade of the backward compatibility. You can't customize the items in back button menu like so:
```jsx
        screenOptions={{
          headerBackTitleVisible: false,
          headerBackTitle: 'Custom Back Title',
        }}
```

![image](https://github.com/react-navigation/react-navigation/assets/39658211/bbbd0043-a805-481c-b0cd-eaa9d9f43316)


But without the backward compatibility you can:

![image](https://github.com/react-navigation/react-navigation/assets/39658211/8004dd1b-81da-4544-8f9e-b9dd05b33026)


What's worse?

Or maybe it is possible to somehow check in JS whether the `backTitleVisible` exist in the `ScreenStackHeaderConfig` and either set the legacy logic or the new one?

Reply:
Yeah let's see if there's a way to detect it

Reply:
I sat on this for a while and I think the safest way to do this would be to add a `export const isNewBackTitleImplementation = true` [here](https://github.com/software-mansion/react-native-screens/blob/main/src/utils.ts#L7) in `react-native-screens` and release it in a 3.21.1 patch.

It's the least hacky way I can think of.

I've added the change in [ffc15a1](https://github.com/react-navigation/react-navigation/pull/11423/commits/ffc15a104a12e924c6d40ad6e575eec6eb561258) and will prepare a PR to screens in a few moments.

Reply:
Here's the PR on the rn-screens side: https://github.com/software-mansion/react-native-screens/pull/1791

Reply:
react-native-screens 3.21.1 got released with the proposed change https://github.com/software-mansion/react-native-screens/releases/tag/3.21.1